sdk/datafactory/Azure.ResourceManager.DataFactory/src/Customized/AvroDataset.cs

```diff
@@ -0,0 +1,193 @@
+        /// <param name="schema"> Columns that define the physical type schema of the dataset. Type: array (or Expression with resultType array), itemType: DatasetSchemaDataElement. </param>
+        /// <param name="linkedServiceName"> Linked service reference. </param>
+        /// <param name="parameters"> Parameters for dataset. </param>
+        /// <param name="annotations"> List of tags that can be used for describing the Dataset. </param>
+        /// <param name="folder"> The folder that this Dataset is in. If not specified, Dataset will appear at the root level. </param>
+        /// <param name="additionalProperties"> Additional Properties. </param>
+        /// <param name="dataLocation">
+        /// The location of the avro storage.
+        /// Please note <see cref="DatasetLocation"/> is the base class. According to the scenario, a derived class of the base class might need to be assigned here, or this property needs to be casted to one of the possible derived classes.
+        /// The available derived classes include <see cref="AmazonS3CompatibleLocation"/>, <see cref="AmazonS3Location"/>, <see cref="AzureBlobFSLocation"/>, <see cref="AzureBlobStorageLocation"/>, <see cref="AzureDataLakeStoreLocation"/>, <see cref="AzureFileStorageLocation"/>, <see cref="FileServerLocation"/>, <see cref="FtpServerLocation"/>, <see cref="GoogleCloudStorageLocation"/>, <see cref="HdfsLocation"/>, <see cref="HttpServerLocation"/>, <see cref="LakeHouseLocation"/>, <see cref="OracleCloudStorageLocation"/> and <see cref="SftpLocation"/>.
+        /// </param>
+        /// <param name="avroCompressionCodec"> The data avroCompressionCodec. Type: string (or Expression with resultType string). </param>
+        /// <param name="avroCompressionLevel"></param>
+        internal AvroDataset(string datasetType, string description, DataFactoryElement<IList<DatasetDataElement>> structure, DataFactoryElement<BinaryData> schema, Core.Expressions.DataFactory.DataFactoryLinkedServiceReference linkedServiceName, IDictionary<string, EntityParameterSpecification> parameters, IList<BinaryData> annotations, DatasetFolder folder, IDictionary<string, BinaryData> additionalProperties, DatasetLocation dataLocation, DataFactoryElement<string> avroCompressionCodec, int? avroCompressionLevel) : base(datasetType, description, structure, null, linkedServiceName, parameters, annotations, folder, additionalProperties)
+        {
+            DataLocation = dataLocation;
+            AvroCompressionCodec = avroCompressionCodec;
+            AvroCompressionLevel = avroCompressionLevel;
+            DatasetType = datasetType ?? "Avro";
+            Schema = schema;
+        }
+
+        /// <summary> Columns that define the physical type schema of the dataset. Type: array (or Expression with resultType array), itemType: DatasetSchemaDataElement. </summary>
+        public new DataFactoryElement<BinaryData> Schema { get; set; }
```

Comment:
You didn't override the parent `Schema` property. Are you sure this is what you want? https://github.com/Azure/azure-sdk-for-net/blob/9ac52a165101b727ff606dd50dd82f771ec7c710/sdk/datafactory/Azure.ResourceManager.DataFactory/src/Generated/Models/DataFactoryDatasetProperties.cs#L65